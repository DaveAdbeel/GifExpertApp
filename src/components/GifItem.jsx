import PropTypes from 'prop-types'
const GifItem = ({ title, url }) => {
  return (
    <li className='card'>
      <img src={url} alt={title} />
      <p data-testid="gif-title">{title}</p>
    </li>
  );
};

export default GifItem;


GifItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}