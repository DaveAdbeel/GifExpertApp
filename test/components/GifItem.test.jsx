/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe("Pruebas en el componente <GifItem/>", () => {
  const id = "JAy6dD3DI78zozppsp";
  const title = "Fabio Porchat Pergunta GIF by Porta Dos Fundos";
  const url =
    "https://media4.giphy.com/media/JAy6dD3DI78zozppsp/200.gif?cid=31efa263g5mm18disiiyzn3kd77tc8f5gqjekntad7n9n1k7&ep=v1_gifs_search&rid=200.gif&ct=g";

  test("El componente deberia tener en las props el id, title y url", (done) => {
    const component = <GifItem id={id} title={title} url={url} />;
    expect(component.props.id).toBeTruthy();

    render(<GifItem id={id} title={title} url={url} />);
    const { alt, src } = screen.getByRole("img");
    expect(alt).toStrictEqual(title);
    expect(src).toStrictEqual(url);

    done();
  });
  test('El componente debe tener un titulo', (done) => { 
    render(<GifItem id={id} title={title} url={url} />);
    expect(screen.getByTestId('gif-title').textContent).toContain(title)

    done()
})
  test("El componente debe hacer match con su snapshot final", (done) => {
    const { container } = render(<GifItem id={id} title={title} url={url} />);
    expect(container).toMatchSnapshot();

    done();
  });
});
