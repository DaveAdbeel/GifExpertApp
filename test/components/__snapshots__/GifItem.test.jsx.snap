// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem/> El componente debe hacer match con su snapshot final 1`] = `
<div>
  <li
    class="card"
  >
    <img
      alt="Fabio Porchat Pergunta GIF by Porta Dos Fundos"
      src="https://media4.giphy.com/media/JAy6dD3DI78zozppsp/200.gif?cid=31efa263g5mm18disiiyzn3kd77tc8f5gqjekntad7n9n1k7&ep=v1_gifs_search&rid=200.gif&ct=g"
    />
    <p
      data-testid="gif-title"
    >
      Fabio Porchat Pergunta GIF by Porta Dos Fundos
    </p>
  </li>
</div>
`;
